packages/next/src/server/base-server.ts

```diff
@@ -2521,16 +2522,17 @@ export default abstract class Server<
             return null
           } catch (err) {
             // If this is during static generation, throw the error again.
-            if (isSSG) throw err
-
-            Log.error(err)
-
             await this.instrumentationOnRequestError(err, req, {
               routerKind: 'App Router',
               routePath: pathname,
               routeType: 'route',
+              revalidateReason: getRevalidateReason(renderOpts),
             })
 
+            if (isSSG) throw err
```

Comment:
nit: the [comment](https://github.com/vercel/next.js/blob/f341e180ab1c548233f9b0293361fd668c1383b3/packages/next/src/server/base-server.ts#L2524) on line 2524 should also be moved to where we are now throwing the error